thinc/loss.py

```diff
@@ -58,30 +58,33 @@ def __init__(
         else:
             self._name_to_i = {}
 
-    def convert_truths(self, truths, guesses: Floats2d) -> Tuple[Floats2d, Floats2d]:
+    def convert_truths(
+        self, truths: IntsOrFloatsOrStrs, guesses: Floats2d
```

Comment:
This doesn't necessarily belong to this PR, but I was at first confused by the `Floats` in `IntsOrFloatsOrStrs` given that we weren't explicitly handling floats in the method below. 

The definition of that type as it stands currently: `IntsOrFloatsOrStrs = Union[Ints1d, Floats2d, Sequence[int], Sequence[str]]`

IMO, the type alias doesn't sufficiently describe the underlying type annotation. I think it would be less confusing to use the latter or rename the former.

Reply:
I completely agree and I agree with @svlandeg as well that the `if-else` forest is a mess, but this PR is trying to make the `CategoricalCrossentropy` more correct than before. And I do feel like there will be more refactor followups.

Reply:
Agreed. I think that applies to my comment breaking this up into functions as well.